### ISSUE TITLE
SignalR's JSON protocol does not support polymorphic serialization on streaming methods

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/aspnetcore#54227]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In #47859, most of ASP.NET Core learned to serialize a polymorphic JSON hierarchy, but SignalR was omitted and still doesn't include the type discriminator declared on the base type, for example [when streaming from server to client](https://github.com/dotnet/aspnetcore/blob/v8.0.2/src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs#L636).

It would be great if this difference would not exist.

For me, the workaround of declaring the derived classes `[JsonPolymorphic]` etc. works.

### Expected Behavior

Streaming a base type from server to client should include the type discriminator.

### Steps To Reproduce

```
public class MyHub: Hub
{
    public async IAsyncEnumerable<BaseMessage> GetMessages()
    {
        yield return new DerivedMessage();
    }
}
```

### .NET Version

8.0.2

**Comment by BrennanConroy** [comment]
Probably dupe of https://github.com/dotnet/aspnetcore/issues/52342
Polymorphic support is added in 9.0, try that.

**Comment by Tragetaschen** [comment]
Yes, this looks very much like a dupe, should've searched the PRs as well…
